go/vt/vtgate/planbuilder/operators/vindex.go

```diff
@@ -81,7 +80,7 @@ func (v *Vindex) AddCol(col *sqlparser.ColName) {
 // checkValid implements the Operator interface
 func (v *Vindex) CheckValid() error {
 	if len(v.Table.Predicates) == 0 {
-		return vterrors.Errorf(vtrpcpb.Code_UNIMPLEMENTED, "unsupported: where clause for vindex function must be of the form id = <val> or id in(<val>,...) (where clause missing)")
+		return vterrors.VT12001("WHERE clause for vindex function must be of the form id = <val> or id in(<val>,...) (where clause missing)")
```

Comment:
Use `vindexUnsupported`.

Reply:
resolved in [f2351bb](https://github.com/vitessio/vitess/pull/10738/commits/f2351bba65f347a8e5ce72ff87757ccaf20115af)